config/software/mingw.rb

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright 2012-2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name "mingw"
+default_version "5.1.0-tdm64-1"
+
+dependency "msys-base"
+dependency "binutils"
+dependency "mingw-runtime"
```

Comment:
msys-base pulls in enough software to run "make" or "patch" or "bash" under msys (i.e. tools that need a posix compatibility layer - a.k.a minimal cygwin but not as bad).  mingw-runtime is the equivalent of "libgcc for windows".  It's the set of libraries compiled for the particular compiler you are bringing down (such as libiconv, libintl, libwinpthread etc.).  These are libraries that we'll need to vendor when compiling ruby and the like.

The long and short of it is - if we choose not to ship devkit in the future, the mingw-runtime libs will need to be vendored, but msys-base will not.  That said, putting all the mingw deps in one place can also be made to work.  I just couldn't figure out an elegant way to fetch from multiple sources in the same software definition and resorting to wget/manually unzipping/manually dealing with caching seemed suboptimal.